src/Controller/ProductController.php

```diff
@@ -36,12 +36,17 @@ public function chargingStations(): Response
     }
 
     /**
-     * Displays informations about the charging stations
-     * @Route("/vehicle", name="vehicle")
+     * Displays informations about vehicle
+     * @Route("/vehicle/{id<^[0-9]+$>}", name="vehicle")
      * @return Response
      */
-    public function vehicle(): Response
+    public function vehicle(VehicleRepository $vehicleRepository, int $id): Response
     {
-        return $this->render('front/product/vehicle.html.twig');
+        return $this->render('front/product/vehicle.html.twig', [
+            'vehicle' => $vehicleRepository->findOneBy(
+                ['id' => $id
+                ]
+            )
+        ]);
```

Comment:
findOneBy with an id should be replaced by find()

Reply:
I will change te findOneBy by find()

Reply:
I forgot that I had already tested with the find (), but it does not return to the vehicle page while with the findOneBy it works.